scripts/parallel_function.R

```diff
@@ -0,0 +1,37 @@
+library(tweedie) 
+library(ggplot2)
+library(foreach)
+library(doParallel)
+
+simTweedieTest <-  
+  function(N){ 
+    t.test( 
+      rtweedie(N, mu=10000, phi=100, power=1.9), 
+      mu=10000 
+    )$p.value 
+  } 
+
+MTweedieTests_parallel <- function(N, M, sig) {
+  cl <- makeCluster(detectCores() - 1)  # create a cluster
+  registerDoParallel(cl)
+  
+  results <- foreach(m=1:M, .combine=c, .packages='tweedie') %dopar% {
+    simTweedieTest(N)
```

Comment:
You should use the .export argument when you are using functions that you have made yourself, in your case the simTweedieTest function.

You should have added the argument .export = "simTweedieTest" in the foreach function.